### ISSUE TITLE
Keras Class Balance error

### ISSUE THREAD

**Issue by andreimouraviev** [fchollet/keras#7112]
Hi there. I'm working on a brain lesion segmentation problem and I'm trying to implement a Unet with code inspired by: https://github.com/jocicmarko/ultrasound-nerve-segmentation

One of the issues I'm trying to overcome is class balance (lots more non-lesion voxels rather than lesion voxels). I tried to use class_weight during the model fit but I'm getting the following error:

_ValueError: class_weight not supported for 3+ dimensional targets._

Does it think that a 512x512 image is 2 dimensional or 262144 (512*512) dimensional. Forgive me if this was explained somewhere else I'm new to keras. I spend a couple of hours hunting down the problem but did not come up with a satisfactory answer. 

Also, if you have advice about how to deal with this problem (like with a different loss function) please let me know.

Basic code to reproduce error: https://gist.github.com/andreimouraviev/fe4222be6ae75b7845d6de07e69bb1cf
